### PR TITLE
SegmentedProgressIndicator composable

### PR DESCRIPTION
**Pull request by garanj**
#### WHAT

Adds support for segmented progress indicators

#### WHY

For some apps, having a progress indicator that can be split into separate segments, indicating, for example, different parts of a workout, can be useful.

#### HOW

Adds the SegmentedProgressIndicator composable.

Example usage:

```
val segments = listOf(
    ProgressIndicatorSegment(1f, Color.Green),
    ProgressIndicatorSegment(1f, Color.Cyan),
    ProgressIndicatorSegment(1f, Color.Magenta),
    ProgressIndicatorSegment(1f, Color.Yellow),
    ProgressIndicatorSegment(2f, Color.Red),
)

SegmentedProgressIndicator(
    modifier = Modifier.fillMaxSize(),
    trackSegments = segments,
    progress = 0.5833f,
    strokeWidth = 10.dp,
    trackColor = Color.Gray,
    paddingAngle = 2f
)
```

#### Checklist :clipboard:
- [ x] Add explicit visibility modifier and explicit return types for public declarations
- [ x] Run spotless check
- [ x] Run tests - **Some other tests failed**
- [ x] Update metalava's signature text files

### CODE REVIEW

**Comment by yschimke** [comment]
Add a Gradle launcher in Android studio for "spotlessApply spotlessCheck compileDebugSources compileReleaseSources metalavaGenerateSignature lintDebug"

It should flag errors locally, and fix up API.

File: composables/src/main/java/com/google/android/horologist/composables/SegmentedProgressIndicator.kt [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+ * represent 3 o'clock, 90 and 180 represent 6 o'clock and 9 o'clock respectively.
+ * By default equal to 270 degrees.
+ * @param strokeWidth The stroke width for the progress indicator.
+ * @param paddingAngle The gap to place between segments. Defaults to 0 degrees.
+ * @param trackColor The background track color. If a segment specifies [trackColor] then the
+ * segment value takes preference. Defaults to [Color.Black]
+ */
+@ExperimentalHorologistComposablesApi
+@Composable
+public fun SegmentedProgressIndicator(
```
Comment (yschimke) [inline composables/src/main/java/com/google/android/horologist/composables/SegmentedProgressIndicator.kt:67]: Can you add one preview for this?

**Comment by garanj** [comment]
> potlessApply spotlessCheck compileDebugSources compileReleaseSources metalavaGenerateSignature lintDebug"

Thanks @yschimke - apologies for not knowing what I'm doing here! :)

File: composables/src/main/java/com/google/android/horologist/composables/SegmentedProgressIndicator.kt (hunk 1 above)
Comment (garanj) [reply composables/src/main/java/com/google/android/horologist/composables/SegmentedProgressIndicator.kt:70]: Added

File: composables/src/main/java/com/google/android/horologist/composables/SegmentedProgressIndicator.kt [hunk 2]
```diff
@@ -106,3 +109,25 @@ public fun SegmentedProgressIndicator(
         remainingProgress -= segmentAngle
     }
 }
+
+@OptIn(ExperimentalHorologistComposablesApi::class)
+@Preview(device = Devices.WEAR_OS_LARGE_ROUND, showSystemUi = true)
```
Comment (yschimke) [inline composables/src/main/java/com/google/android/horologist/composables/SegmentedProgressIndicator.kt:114]: nit: Maybe use https://github.com/google/horologist/blob/main/compose-tools/src/main/java/com/google/android/horologist/compose/tools/WearPreviewDevices.kt

To show on square also?

@WearSquareDevicePreview

**Comment by garanj** [comment]
> One thing I wonder about this, is whether we should use more of the existing defaults? from the CircularProgressIndicator in wear compose?
> 
> ```
> trackColor: Color = MaterialTheme.colors.onBackground.copy(alpha = 0.1f),
> strokeWidth: Dp = ProgressIndicatorDefaults.StrokeWidth,
> ```

Great point, agreed, I'll take a look

**Comment by yschimke** [comment]
@garanj For the screenshot, are there other samples you could use with a more designer-y colour pallete?

File: composables/src/main/java/com/google/android/horologist/composables/SegmentedProgressIndicator.kt (hunk 2 above)
Comment (garanj) [reply composables/src/main/java/com/google/android/horologist/composables/SegmentedProgressIndicator.kt:136]: I added square preview but was unable to get `WearSquareDevicePreview` annotation to actually trigger the preview in AS. Maybe I need to update Android Studio or something.

**Review by yschimke — APPROVED** [review]

File: composables/src/main/java/com/google/android/horologist/composables/SegmentedProgressIndicator.kt (hunk 2 above)
Comment (yschimke) [reply composables/src/main/java/com/google/android/horologist/composables/SegmentedProgressIndicator.kt:136]: Possibly.

**Pull request merged**